src/tmt_web/service.py

```diff
@@ -0,0 +1,191 @@
+import logging
+import os
+
+import tmt
+from celery.app import Celery  # type: ignore[attr-defined]
+from tmt.utils import Path  # type: ignore[attr-defined]
+
+from tmt_web.generators import html_generator as html
+from tmt_web.generators import json_generator, yaml_generator
+from tmt_web.utils import git_handler as utils
+
+logger = tmt.Logger(logging.getLogger("tmt-logger"))
+
+redis_url = os.getenv("REDIS_URL", "redis://localhost:6379")
+
+app = Celery(__name__, broker=redis_url, backend=redis_url)
+
+
+def get_tree(url: str, name: str, ref: str | None, tree_path: str) -> tmt.base.Tree:
+    """
+    This function clones the repository and returns the Tree object
+    :param ref: Object ref
+    :param name: Object name
+    :param url: Object url
+    :param tree_path: Object path
+    :return:
+    """
+    logger.print("Cloning the repository for url: " + url)
+    logger.print("Parsing the url and name...")
+    logger.print("URL: " + url)
+    logger.print("Name: " + name)
+
+    path = utils.get_git_repository(url, logger, ref)
+
+    if tree_path is not None:
+        tree_path += '/'
+        # If path is set, construct a path to the tmt Tree
+        if path.suffix == '.git':
+            path = path.with_suffix('')
+        path = Path(path.as_posix() + tree_path)
+
```

Comment:
```suggestion
    if tree_path:
        tree_path = f"{tree_path}/"
        # If path is set, construct a path to the tmt Tree
        path = Path(f"{path.with_suffix('').as_posix()}{tree_path}")  
```
It's always without suffix, right? i.e. we don't need to check for `.git` specifically?

Reply:
Currently the path can both include the suffix and not include it. I guess it will be solved by #9 anyway.